functions/get-redeem-code.ts

```diff
@@ -0,0 +1,93 @@
+import { getGiftCardOrderId, getMessageToSign } from "../shared/helpers";
+import { ReloadlyFailureResponse, ReloadlyRedeemCodeResponse } from "./types";
+import { verifyMessage } from "ethers/lib/utils";
+import { getTransactionFromOrderId } from "./get-order";
+import { validateEnvVars, validateRequestMethod } from "./validators";
+import { Env, getAccessToken, getBaseUrl, commonHeaders } from "./helpers";
+import { AccessToken } from "./types";
+
+export const onRequest: PagesFunction<Env> = async (ctx) => {
+  try {
+    validateRequestMethod(ctx.request.method, "GET");
+    validateEnvVars(ctx);
+
+    const accessToken = await getAccessToken(ctx.env);
+
+    const { searchParams } = new URL(ctx.request.url);
+    const transactionId = Number(searchParams.get("transactionId"));
+    const signedMessage = searchParams.get("signedMessage");
+    const wallet = searchParams.get("wallet");
+    const permitSig = searchParams.get("permitSig");
+
+    if (isNaN(transactionId) || !(transactionId && signedMessage && wallet && permitSig)) {
+      throw new Error(
+        `Invalid query parameters: ${{
+          transactionId,
+          signedMessage,
+          wallet,
+          permitSig,
+        }}`
+      );
+    }
+
+    const errorResponse = Response.json({ message: "Given details are not valid to redeem code." }, { status: 403 });
+
+    if (verifyMessage(getMessageToSign(transactionId), signedMessage) != wallet) {
+      console.error(
+        `Signed message verification failed: ${JSON.stringify({
+          signedMessage,
+          transactionId,
+        })}`
+      );
+      return errorResponse;
+    }
+
+    const orderId = getGiftCardOrderId(wallet, permitSig);
+    const order = await getTransactionFromOrderId(orderId, accessToken);
+
+    if (order.transactionId != transactionId) {
+      console.error(
+        `Given transaction does not match with retrieved transactionId using generated orderId: ${JSON.stringify({
+          transactionId,
+          orderId,
+          transactionIdFromOrder: order.transactionId,
+        })}`
+      );
+      return errorResponse;
+    }
+
+    const redeemCode = await getRedeemCode(transactionId, accessToken);
+    return Response.json(redeemCode, { status: 200 });
+  } catch (error) {
+    console.error("There was an error while processing your request.", error);
+    return Response.json({ message: "There was an error while processing your request." }, { status: 500 });
+  }
+};
+
+export const getRedeemCode = async (transactionId: number, accessToken: AccessToken) => {
```

Comment:
Arrow function syntax shouldn't pass linter rules. Please change to named functions!

Reply:
I see we use named functions and I wanted to keep it that way. But wrangler was providing better types support using arrow functions. That is why I kept all backend functions as arrow functions. But I am going to try again and turn them into named functions.